semgrep/tests/e2e/test_check.py

```diff
@@ -12,6 +14,27 @@
 )
 
 
+def etree_to_dict(t):
```

Comment:
This is a nice little snippet I've used a few times. Taken from: https://stackoverflow.com/a/10077069

It's not a perfect XML <-> Dict/JSON, but it's sufficient for this simple usage :+1:

Reply:
maybe add this note in a comment then? (the link to stackoverflow) and that the alternative was to use xmldiff but it adds an annoying dependency?

Reply:
Eh, generally I subscribe more to the ["Don’t comment bad code -- rewrite it"](https://www.goodreads.com/quotes/857949-don-t-comment-bad-code-rewrite-it) philosophy, as opposed to lots of comments, but I can add a reference to the stackoverflow post. If anyone needs more context in the future they can always come see this conversation, after all.

Reply:
I prefer to add comments about design decisions in the code; git blame + git log + jump to github is 3 hops away to get the information.

Reply:
in 2 months someone can come here and replace your code by xmldiff because it does not know what was the problem with xmldiff in the first place.

Reply:
I also like "dont comment bad code -- rewrite it", but that does not prevent to add comments in the new good code about  alternative approach and what was wrong with those alternatives approach.